ballerina/read.bal

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+import ballerina/lang.'object as obj;
+
+public type JsonPathValue json;
+
+public type JsonPathRawTemplate object {
+    *obj:RawTemplate;
+    public string[] & readonly strings;
+    public JsonPathValue[] insertions;
+};
+
+# Extract details from the given JSON value using the provided query template expression
+# + 'json - JSON value
+# + query - JSON path expression
+# + return - extracted details as JSON value, a jsonpath:Error otherwise
+public isolated function read(json 'json, JsonPathRawTemplate query) returns json|Error {
```

Comment:
Shall we add few examples in here? Otherwise when we generate there are no examples for the user to refer on how to use this api.

Reply:
+1, will add